File: programmers/algorithm/cpp/완주하지 못한 선수.cc
#include <iostream>
#include <algorithm>
#include <string>
#include <vector>
#include <unordered_map>
#define endl '\n'
using namespace std;
unordered_map<string, int> m;

string solution(vector<string> participant, vector<string> completion) {
    
    for (string str: participant) {
        if (m.find(str) != m.end()) {
            m[str] = m[str] + 1;
        } else {
            m[str] = 1;
        }
    }
    
    for (string part: completion) {
        m[part] = m[part] - 1;
    }
    
    
    
    
    for (auto entry: m) {
        if (entry.second != 0)
            return entry.first;
    }
    
    return "";
}

File: programmers/algorithm/java/완주하지 못한 선수.java
import java.util.*;


class Solution {
    static HashMap<String, Integer> hm = new HashMap<>();
    
    public String solution(String[] participant, String[] completion) {
        String res = "";
        
        for (int i = 0; i < participant.length; ++i) {
            String key = participant[i];
            hm.put(key, hm.getOrDefault(key, 0) + 1);
        }
         
        for (int i = 0; i < completion.length; ++i) {
            String key = completion[i];
            hm.put(key, hm.get(key) - 1);
        }
        
        
        for (Map.Entry<String, Integer> entry: hm.entrySet()) {
            if (entry.getValue() != 0)
                res = entry.getKey();
        }
        
        return res;
    }
}